### PR TITLE
docs-requirements.txt: Adds sphinx-argparse

### PR DESCRIPTION
**Pull request by BrateS**
Adds the required module for the documentation page to work.
docs/Developers/Documentation_coala_Settings.rst: Adds the documentation page

Uses the module installed to generate the documentation page
using the default_arg_parser function.
docs/conf.py: Adds the extension

Adds the extension for sphinx in the configuration file
docs/index.rst:Adds Doc page in index
Adds the documentation page in the index file so it appears.

Fixes: https://github.com/coala/documentation/issues/40

### Checklist

- [ ] I have rebased properly. Please see [our tutorial on rebasing](http://coala.io/git#rebasing).
- [ ] I have gone through the [commit guidelines](http://coala.io/commit) and I've followed them.
- [ ] I have followed the [guidelines for the commit shortlog](http://coala.io/commit#shortlog).
- [ ] I have followed the [guidelines for the commit body](http://coala.io/commit#commit-body).
- [ ] I have included the issue URL with the 'Fixes' keyword if the commit fixes a *real bug* and the 'Closes' keyword if it adds a feature or enhancement. For details, check the [issue reference section in our commit guidelines](http://coala.io/commit#issue-reference).
- [ ] I have [run all the tests](http://api.coala.io/en/latest/Developers/Executing_Tests.html) and they all pass. If any CI (below) is not green, please fix them. If GitMate issues appear you will have to amend your commits, pushing new commits on top is not sufficient!

### CODE REVIEW

File: docs/Developers/Doc_coala_Settings.rst [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Documentation for all coala settings
+====================================
```
Comment (sils) [inline docs/Developers/Doc_coala_Settings.rst]: add newline here

File: docs/index.rst [hunk 2]
```diff
@@ -29,6 +29,7 @@ If you're trying to **use** coala, you should have a look at our
    :hidden:
 
    Newcomers' Guide <Developers/Newcomers_Guide>
+   Documentation coala settings <Developers/Doc_coala_Settings>
```
Comment (sils) [inline docs/index.rst]: can we have the link title == document title?

Comment (sils) [inline docs/index.rst]: also I dislike the filename, maybe just coala_settings.rst and the title could be `coala Settings`

**Comment by sils** [comment]
Hi,

it looks like your PR is out of date and needs a rebase. [This page](http://coala.io/git) may help you to get started on this. (*Do not press the update branch button. It's evil.*)

Add the original coala repository as a remote:

```
git remote add upstream https://github.com/coala-analyzer/coala.git
```

(Be sure to use coala-bears on the coala-bears repository :))

Then go to your branch and rebase it onto our master:

```
git checkout <yourbranch>
git fetch upstream
git rebase --interactive upstream/master
```

Now push it up. You will need to do a force push because you change history here (do not do this on master branches :)):

```
git push --force
```

If you're just looking to fix an issue very quickly and not interested in contributing to coala long term, we can fix it up for you - just tell us :).

Ping us on [our gitter channel](https://coala.io/chat) if you need help!

**Comment by sils** [comment]
remove the colon after closes please

**Comment by sils** [comment]
otherwise it seems good, I wanna have it ;) we can then look into patching visuals upstream

**Comment by sils** [comment]
oh sorry, Adds -> Add in shortlog. That's like a death sin :P

**Comment by BrateS** [comment]
@sils Done,

**Comment by sils** [comment]
it says still Adds

**Comment by BrateS** [comment]
@sils Like that ?

**Comment by sils** [comment]
yep!

**Comment by sils** [comment]
ack https://github.com/coala/coala/pull/3315/commits/9136c0380de943ec20bcb79076c2955b9ebc9cd8

**Comment by sils** [comment]
@rultor merge

**Comment by sils** [comment]
*so excited*

**Pull request merged**